syntax = "proto3";

package search;

service SearchService {
  rpc SayHello (HelloRequest) returns (HelloResponse);
  rpc Query (QueryRequest) returns (QueryResponse);
  rpc StreamQuery (stream QueryRequest) returns (stream QueryResponse);
}

message HelloRequest {
  string name = 1;
}

message HelloResponse {
  string message = 1;
}

message QueryRequest {
  string query = 1;
  int32 page = 2;
  int32 limit = 3;
}

message Document {
    string name = 1;
    string location = 2;
}

message QueryResponse {
  int32 total_hits = 1;
  repeated Document items = 2;
  
}
